pkg/offline/offline.go

```diff
@@ -77,40 +78,81 @@ func WithQueue(filepath string, syncLimit int) (heartbeat.HandleOption, error) {
 	}, nil
 }
 
-// Sync returns a function to send queued heartbeats to the WakaTime API.
+// Sync returns a function to send all queued heartbeats to the WakaTime API.
 func Sync(filepath string, syncLimit int) func(next heartbeat.Handle) error {
 	return func(next heartbeat.Handle) error {
 		log.Debugf("execute offline sync with file %s", filepath)
 
-		hh, err := popHeartbeats(filepath, syncLimit)
-		if err != nil {
-			return fmt.Errorf("failed to fetch heartbeat from offline queue: %s", err)
-		}
+		var (
+			count   int
+			exclude []string
+		)
 
-		if len(hh) == 0 {
-			log.Debugln("abort execution, as there are no queued heartbeats ready for sending")
+		for {
+			if count > maxSyncRequests {
+				return fmt.Errorf("exceeded max sync request limit of %d", maxSyncRequests)
+			}
 
-			return nil
-		}
+			opts := PopOptions{
+				Excluded: exclude,
+				Limit:    syncLimit,
```

Comment:
Does `syncLimit` come from `--sync-offline-activity syncLimit`? That param should control the total heartbeats sent, but not how many we send per request. Per request, we should always send `<= 24` heartbeats.

Reply:
Ah alright, I didn't know this. Valuable info! Will adjust :+1: